xds/src/test/java/io/grpc/xds/GrpcXdsClientImplDataTest.java

```diff
@@ -2342,6 +2344,113 @@ public void parseCluster_validateEdsSourceConfig() throws ResourceInvalidExcepti
         LoadBalancerRegistry.getDefaultRegistry());
   }
 
+  @Test
+  public void processCluster_parsesMetadata()
+      throws ResourceInvalidException, InvalidProtocolBufferException {
+    ClusterMetadataRegistry testRegistry = ClusterMetadataRegistry.getInstance();
+    testRegistry.registerParser(new ClusterMetadataRegistry.ClusterMetadataValueParser() {
+      @Override
+      public String getTypeUrl() {
+        return "type.googleapis.com/test.Type";
+      }
+
+      @Override
+      public Object parse(Any value) {
+        return "testValue";
+      }
+    });
+
+    Any typedFilterMetadata = Any.newBuilder()
+        .setTypeUrl("type.googleapis.com/test.Type")
+        .setValue(ByteString.copyFromUtf8("test"))
+        .build();
+
+    Struct filterMetadata = Struct.newBuilder()
+        .putFields("key1", Value.newBuilder().setStringValue("value1").build())
```

Comment:
Should we craft other types of Values in the test cases to excersise all branches of the `convertValue` method above (e.g. `LIST_VALUE`, `BOOL_VALUE`, etc.)?

Reply:
I don't feel like doing so right now, as we are concerned only about Audience here which will always be StringValue. We definitely need to add tests as and when the usage grows. Adding more tests on different value types right now will be purely based on assumption.

Reply:
> we are concerned only about Audience here which will always be StringValue ... Adding more tests on different value types right now will be purely based on assumption.

If we not using the other cases and only using `STRING_VALUE`, then we should delete the other unused branches in `convertValue` until they are needed. This would keep the code and tests focused.

Reply:
Independent of the plan, I suspect we shouldn't do that sort of testing here. It can be part of a separate ProtobufJsonConverterTest.

Deleting the branches in ProtobufJsonConverter doesn't seem to benefit anyone.